lib/serverless-typescript-demo-stack.ts

```diff
@@ -33,20 +24,40 @@ export class ServerlessTypescriptDemoStack extends Stack {
       removalPolicy: RemovalPolicy.DESTROY
     });
 
+    const envVariables = {
+      AWS_ACCOUNT_ID: Stack.of(this).account,
+      POWERTOOLS_SERVICE_NAME: 'serverless-typescript-demo',
+      POWERTOOLS_LOGGER_LOG_LEVEL: 'WARN',
+      POWERTOOLS_LOGGER_SAMPLE_RATE: '0.01',
+      POWERTOOLS_LOGGER_LOG_EVENT: 'true',
+      POWERTOOLS_METRICS_NAMESPACE: 'AwsSamples',
+  };
+
+    const esBuildSettings = {
+      minify: true,
+      platform: "node"
```

Comment:
[platform=node](https://github.com/aws/aws-cdk/blob/v1.165.0/packages/%40aws-cdk/aws-lambda-nodejs/lib/bundling.ts#L183-L184) is the default and I don't think BundlerSettings#platform is read, it's not a [documented option.](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-lambda-nodejs.BundlingOptions.html)

Reply:
I recommend removing `"strictPropertyInitialization": false` and the other `strict` overrides from tsconfig. 
Then add `BundlingOptions` type annotation to L36 and you should get a type error on platform. 

https://github.com/aws-samples/serverless-typescript-demo/blob/main/tsconfig.json#L21